launchpod/angular-launchpod/backend/src/test/java/com/google/launchpod/LoginServletTest.java

```diff
@@ -0,0 +1,180 @@
+import java.security.interfaces.DSAKey;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import com.google.launchpod.data.LoginStatus;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.launchpod.servlets.LoginServlet;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * Runs unit tests for the LoginServlet that contains doGet() method.
+ */
+@RunWith(JUnit4.class)
+public class LoginServletTest extends Mockito {
+
+  @InjectMocks
+  private LoginServlet servlet = new LoginServlet();
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Rule // JUnit 4 uses Rules for testing specific messages
+  public ExpectedException thrown = ExpectedException.none();
+
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalUserServiceTestConfig());
+
+  private static final Gson GSON = new Gson();
+  JsonParser parser = new JsonParser();
+
+  private static final String EMAIL = "email";
+
+  private static final String TEST_EMAIL = "123@abc.com";
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Asserts that doPost() gets the user's correct status when logged in.
+   */
+  @Test
+  public void doGet_GetsCorrectStatusLoggedIn() throws IOException {
+    helper.setEnvIsLoggedIn(true).setEnvEmail(TEST_EMAIL).setEnvAuthDomain("localhost");
+    UserService userService = UserServiceFactory.getUserService();
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    String urlToRedirectToAfterUserLogsOut = "/index.html";
+    String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    String loginMessage = "<p>Logged in as " + TEST_EMAIL + ". <a href=\"" + logoutUrl + "\">Logout</a>.</p>";
+    LoginStatus loginStatus = new LoginStatus(true, loginMessage);
+    verify(response).setContentType("application/json");
+    assertEquals(parser.parse(GSON.toJson(loginStatus.isLoggedIn)), parser.parse(stringWriter.toString()).getAsJsonObject().get("isLoggedIn"));
```

Comment:
i think it's easier to do: 
 assertEquals(loginStatus.isLoggedIn, GSON.fromJson(stringWriter.toString(), LoginStatus.class).isLoggedIn);

it avoids serialization of loginStatus

Reply:
also do for other tests

Reply:
done.